src/test/compile-fail/use-mod-4.rs

```diff
@@ -11,4 +11,7 @@
 use foo::self; //~ ERROR unresolved import `foo::self`
```

Comment:
I don't think it makes sense to improve errors in this area, all examples from https://github.com/rust-lang/rust/issues/42559 should be legal anyway, I'll try to submit a PR somewhere after June 17.